pkg/apis/autoscaling/v2beta1/conversion.go

```diff
@@ -0,0 +1,399 @@
+		Convert_v2beta1_ObjectMetricStatus_To_autoscaling_ObjectMetricStatus,
+		Convert_autoscaling_PodsMetricStatus_To_v2beta1_PodsMetricStatus,
+		Convert_v2beta1_PodsMetricStatus_To_autoscaling_PodsMetricStatus,
+		Convert_autoscaling_ResourceMetricSource_To_v2beta1_ResourceMetricSource,
+		Convert_v2beta1_ResourceMetricSource_To_autoscaling_ResourceMetricSource,
+		Convert_autoscaling_MetricTarget_To_v2beta1_CrossVersionObjectReference,
+		Convert_v2beta1_CrossVersionObjectReference_To_autoscaling_MetricTarget,
+		Convert_autoscaling_ResourceMetricStatus_To_v2beta1_ResourceMetricStatus,
+		Convert_v2beta1_ResourceMetricStatus_To_autoscaling_ResourceMetricStatus,
+		Convert_autoscaling_HorizontalPodAutoscaler_To_v2beta1_HorizontalPodAutoscaler,
+		Convert_v2beta1_HorizontalPodAutoscaler_To_autoscaling_HorizontalPodAutoscaler,
+	)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func Convert_autoscaling_MetricTarget_To_v2beta1_CrossVersionObjectReference(in *autoscaling.MetricTarget, out *autoscalingv2beta1.CrossVersionObjectReference, s conversion.Scope) error {
+	return nil
+}
+
+func Convert_v2beta1_CrossVersionObjectReference_To_autoscaling_MetricTarget(in *autoscalingv2beta1.CrossVersionObjectReference, out *autoscaling.MetricTarget, s conversion.Scope) error {
+	return nil
+}
+
+func Convert_v2beta1_ResourceMetricStatus_To_autoscaling_ResourceMetricStatus(in *autoscalingv2beta1.ResourceMetricStatus, out *autoscaling.ResourceMetricStatus, s conversion.Scope) error {
+	out.Name = core.ResourceName(in.Name)
+	utilization := in.CurrentAverageUtilization
+	averageValue := in.CurrentAverageValue
+	out.Current = autoscaling.MetricValueStatus{
+		AverageValue:       &averageValue,
+		AverageUtilization: utilization,
+	}
+	return nil
+}
+
+func Convert_autoscaling_ResourceMetricStatus_To_v2beta1_ResourceMetricStatus(in *autoscaling.ResourceMetricStatus, out *autoscalingv2beta1.ResourceMetricStatus, s conversion.Scope) error {
+	out.Name = v1.ResourceName(in.Name)
+	out.CurrentAverageUtilization = in.Current.AverageUtilization
+	if in.Current.AverageValue != nil {
+		out.CurrentAverageValue = *in.Current.AverageValue
+	}
+	return nil
+}
+
+func Convert_v2beta1_ResourceMetricSource_To_autoscaling_ResourceMetricSource(in *autoscalingv2beta1.ResourceMetricSource, out *autoscaling.ResourceMetricSource, s conversion.Scope) error {
+	out.Name = core.ResourceName(in.Name)
+	utilization := in.TargetAverageUtilization
+	averageValue := in.TargetAverageValue
+	out.Target = autoscaling.MetricTarget{
+		AverageValue:       averageValue,
+		AverageUtilization: utilization,
+	}
+	return nil
+}
+
+func Convert_autoscaling_ResourceMetricSource_To_v2beta1_ResourceMetricSource(in *autoscaling.ResourceMetricSource, out *autoscalingv2beta1.ResourceMetricSource, s conversion.Scope) error {
+	out.Name = v1.ResourceName(in.Name)
+	out.TargetAverageUtilization = in.Target.AverageUtilization
+	out.TargetAverageValue = in.Target.AverageValue
```

Comment:
you need to persist selector, too.

Reply:
`ResourceMetricSource` doesn't have a selector in the new version or v2beta1, was that overlooked?

Reply:
Yeah, you'll need to add a selector field.  New API versions are weird.

Reply:
(to the old version)

Reply:
My point is, there isn't a selector field in the new version to persist to the old version. The KEP only adds `Target` to Resource metrics. I'm assuming that was intentional. Did you want to add metric selectors to Resource metrics as well?